File: sha256/python/main.py
import sys
import hashlib  # use hashlib instead of sha256 module

def read_file(file_path):
    try:
        with open(file_path, 'rb') as file:
            return file.read()
    except FileNotFoundError:
        print(f"Error: File not found - {file_path}")
        sys.exit(1)

if len(sys.argv) != 2:
    print(f"Usage: {sys.argv[0]} <file_path>")
    sys.exit(1)

file_path = sys.argv[1]
file_content = read_file(file_path)
hash = hashlib.sha256(file_content).hexdigest()[:8]  # Truncate hash to first 8 characters
print(hash)

File: sha256/cpp/main.cpp
// main.cpp
// Demonstrates usage of our SHA256 hashing functions

#include "sha256.h"
#include <iostream>
#include <fstream>
#include <string>

bool fileExists(const std::string& filePath) {
    std::ifstream file(filePath);
    return file.good();
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        std::cerr << "Usage: " << argv[0] << " <file_path>" << std::endl;
        return 1;
    }

    std::string file_path = argv[1];

    if (!fileExists(file_path)) {
        std::cerr << "Error: File not found - " << file_path << std::endl;
        return 1;
    }

    try {
        std::string hash = shortHashFile(file_path);
        std::cout << hash << std::endl;
    } catch (const std::exception& e) {
        std::cerr << e.what() << std::endl;
        return 1;
    }

    return 0;
}
